palette/src/lch.rs

```diff
@@ -312,126 +314,41 @@ where
 {
     type Scalar = T;
 
-    #[rustfmt::skip]
     fn get_color_difference(&self, other: &Lch<Wp, T>) -> Self::Scalar {
         let self_a = clamp(
             self.chroma.max(T::zero()) * self.hue.to_radians().cos(),
             from_f64(-128.0),
-            from_f64(127.0)
+            from_f64(127.0),
         );
         let self_b = clamp(
             self.chroma.max(T::zero()) * self.hue.to_radians().sin(),
             from_f64(-128.0),
-            from_f64(127.0)
+            from_f64(127.0),
         );
         let other_a = clamp(
             other.chroma.max(T::zero()) * other.hue.to_radians().cos(),
             from_f64(-128.0),
-            from_f64(127.0)
+            from_f64(127.0),
         );
         let other_b = clamp(
             other.chroma.max(T::zero()) * other.hue.to_radians().sin(),
             from_f64(-128.0),
-            from_f64(127.0)
+            from_f64(127.0),
         );
-        let c_bar = (self.chroma + other.chroma) / from_f64(2.0);
-        let c_bar_pow_seven = c_bar * c_bar * c_bar * c_bar * c_bar * c_bar * c_bar;
-        let twenty_five_pow_seven = from_f64(6103515625.0);
-        let pi_over_180 = from_f64::<T>(core::f64::consts::PI / 180.0);
-
-        let g = from_f64::<T>(0.5)
-            * (from_f64::<T>(1.0)
-                - (c_bar_pow_seven / (c_bar_pow_seven + twenty_five_pow_seven)).sqrt());
-        let a_one_prime = self_a * (from_f64::<T>(1.0) + g);
-        let a_two_prime = other_a * (from_f64::<T>(1.0) + g);
-        let c_one_prime = (a_one_prime * a_one_prime + self_b * self_b).sqrt();
-        let c_two_prime = (a_two_prime * a_two_prime + other_b * other_b).sqrt();
-
-        let calc_h_prime = |b: T, a_prime: T| -> T {
-            if b == T::zero() && a_prime == T::zero() {
-                from_f64(0.0)
-            } else {
-                let result = b.atan2(a_prime).to_degrees();
-                if result < T::zero() {
-                    result + from_f64(360.0)
-                } else {
-                    result
-                }
-            }
-        };
-        let h_one_prime = calc_h_prime(self_b, a_one_prime);
-        let h_two_prime = calc_h_prime(other_b, a_two_prime);
-
-        let h_prime_difference = (h_one_prime - h_two_prime).abs();
-
-        let delta_h_prime: T = if c_one_prime == T::zero() || c_two_prime == T::zero() {
-            from_f64(0.0)
-        } else {
-            if h_prime_difference <= from_f64(180.0) {
-                h_two_prime - h_one_prime
-            } else {
-                if h_two_prime <= h_one_prime {
-                    h_two_prime - h_one_prime + from_f64(360.0)
-                } else {
-                    h_two_prime - h_one_prime - from_f64(360.0)
-                }
-            }
+        let self_params = ColorDiff {
+            l: self.l,
+            a: self_a,
+            b: self_b,
+            chroma: (self_a * self_a + self_b * self_b).sqrt(),
```

Comment:
This is only different from `self.chroma` because of the clamping, right?

Reply:
That was an error copying and pasting the reusable code from `lab.rs`. I wasn't using that for the Lch "preparation". It is correct that it should be `chroma: self.chroma`.

I've corrected it back.

Reply:
Ok, I thought it looked a bit odd. But nice to get rid of a square root, at least.